qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -0,0 +1,161 @@
+        other_unique = {key: other._data[key] for key in other._data.keys() - self._data.keys()}
+        new_data.update(other_unique)
+
+        register_length = max(self.register_length, other.register_length)
+
+        return self.__class__(new_data, register_length, copy=False)
+
+    def _multiply(self, other: complex) -> SparseLabelOp:
+        """Return scalar multiplication of self and other.
+
+        Args:
+            other: the number to multiply the ``SparseLabelOp`` values by.
+
+        Returns:
+            The newly multiplied ``SparseLabelOp``.
+
+        Raises:
+            TypeError: if ``other`` is not compatible type (int, float or complex)
+        """
+        if not isinstance(other, Number):
+            raise TypeError(
+                f"Unsupported operand type(s) for *: 'SparseLabelOp' and '{type(other).__name__}'"
+            )
+        new_data = {key: val * other for key, val in self._data.items()}
+
+        return self.__class__(new_data, self.register_length, copy=False)
+
+    def conjugate(self) -> SparseLabelOp:
+        """Returns the conjugate of the ``SparseLabelOp``.
+
+        Returns:
+            The complex conjugate of the starting ``SparseLabelOp``.
+        """
+        new_data = {key: np.conjugate(val) for key, val in self._data.items()}
+
+        return self.__class__(new_data, self.register_length, copy=False)
+
+    @abstractmethod
+    def transpose(self) -> SparseLabelOp:
+        """Returns the transpose of the ``SparseLabelOp``.
+
+        Returns:
+            The transpose of the starting ``SparseLabelOp``.
+        """
+
+    def equiv(self, other: SparseLabelOp) -> bool:
+        """Check equivalence of two ``SparseLabelOp`` instances up to an accepted tolerance.
+
+        The absolute and relative tolerances can be changed via the `atol` and `rtol` attributes,
+        respectively.
+
+        Args:
+            other: the second ``SparseLabelOp`` to compare with this instance.
+
+        Returns:
+            True if operators are equivalent, False if not.
+        """
+        if not isinstance(other, SparseLabelOp):
+            return False
+        if self.register_length != other._register_length:
+            return False
+        if self._data.keys() != other._data.keys():
+            return False
+        for key, value in self._data.items():
+            if not cmath.isclose(value, other._data[key], rel_tol=self.rtol, abs_tol=self.atol):
+                return False
+        return True
+
+    def __eq__(self, other: object) -> bool:
+        """Check exact equality of two ``SparseLabelOp`` instances
+
+        Args:
+            other: the second ``SparseLabelOp`` to compare with this instance.
+
+        Returns:
+            True if operators are equal, False if not.
+        """
+        if not isinstance(other, SparseLabelOp):
+            return False
+        return self.register_length == other._register_length and self._data == other._data
```

Comment:
register_length here is using the getter on self and the private var on other. I think we should be consistent one way or the other